ui/app/src/components/GitManagement/PublishCommitDialog/PublishCommitDialog.tsx

```diff
@@ -89,23 +88,26 @@ export function PublishCommitDialog(props: PublishCommitDialogProps) {
       });
     }
   };
-  useMount(() => {
-    setState({ loadingPublishingTargets: true });
-    const sub = fetchPublishingTargets(site).subscribe({
-      next({ publishingTargets }) {
-        const newData: Partial<PublishCommitDialogState> = { publishingTargets, loadingPublishingTargets: false };
-        // Set pre-selected publishing target.
-        if (publishingTargets.length > 0) {
-          const stagingEnv = publishingTargets.find((target) => target.name === 'staging');
-          newData.publishingTarget = stagingEnv?.name ?? publishingTargets[0].name;
+  useEffect(() => {
+    let sub;
```

Comment:
Move the let and the destruction function inside the if. You may loose the optional chaining of sub in the destruct fn.

Reply:
You didn't move the sub inside...